packages/game-client/src/App.jsx

```diff
@@ -143,7 +150,7 @@ function App() {
 					updatedHash[newHeadKey] = newHead;
 					updatedList.unshift(newHeadKey);
 				}
-				if (newHeadKey in prevSnakes[snakeId].hash) {
+				if (newHeadKey in prevSnakes[snakeId].hash || collisionDetection(newHeadKey,prevSnakes)) {
```

Comment:
Break this up into separate if else statements... 
} else if(collisionDetection(newHeadKey, prevSnakes) {}
This will help us log how did the player die... move newHeadKey in prevSnakes[snakeId].hash this logic into a function as well... Name the functions correctly... collisionDetection the word doesn't clarify to what the snake has collided... A wall or another player?